### ISSUE TITLE
Add locl OT feature for Lithuanian

### ISSUE THREAD

**Issue by MariannaPaszkowska** [googlefonts/googlesans-flex#280]
If the fonts support dynamic mark positioning for composing diacritics, consider adding

- [ ] locl (Latin: Catalan, Romanian, Moldovan, Dutch, **Lithuanian**)

The convention in Lithuanian is not to suppress the dot on _i_ (and by extension _j_) when marks are added above, as may occur in dictionaries.

_Originally posted by @tiroj in https://github.com/googlefonts/googlesans-flex/issues/33#issuecomment-1311104692_

**Comment by MariannaPaszkowska** [comment]
We could perhaps use the following:
```
script latn;
language LTH ;
sub idotless' @CombiningTopAccents by i;
```
```
@CombiningTopAccents = [gravecomb acutecomb circumflexcomb tildecomb macroncomb brevecomb dotaccentcomb dieresiscomb ringcomb hungarumlautcomb caroncomb commaturnedabovecomb] 
```
Recommendation from the [Unicode standard v 11.0](https://www.unicode.org/versions/Unicode11.0.0/ch07.pdf#G15992) is slightly different:
```
To express the forms sometimes used in the Baltic (where the dot is retained under a top
accent in dictionaries), use i + overdot + accent (see Figure 7-2).

```
@tiroj How would you recommend the implementation to be here? Is the overdot mentioned by the standard the U+0307 glyph (Combining Dot Above)?

**Comment by tiroj** [comment]
Yes, Unicode recommends that Lithuanian text with above accents stack them above U+0307, which is more robust than relying on language-specific glyph processing. It isn’t clear, however, whether Lithuanian text can be relied upon to be encoded in that way, so we tend to provide the locl behaviour as a fallback.

**Comment by MariannaPaszkowska** [comment]
Thank you, @tiroj, this is very helpful!

**Comment by chrissimpkins** [comment]
@MariannaPaszkowska @dberlow re: @moyogo's comment https://github.com/googlefonts/googlesans-flex/issues/33#issuecomment-1342213139 and @tiroj's comment in https://github.com/googlefonts/googlesans-flex/issues/280#issuecomment-1409279465, do we currently support the combining dot above if the text is encoded in this way?  I'm trying to understand if this is an issue for our Lithuanian support.

**Comment by moyogo** [comment]
I’d recommened again not to insert any dot that is not there, given the scope of the font it would be more destructive than positive. But if you still want to do this, at least do it properly for the characters that would actually benefit from it and not others. Only acute, grave, tilde for i, acute and tilde for iogonek, and tilde for j are used in Lithuanian dictionary notation.

> Yes, Unicode recommends that Lithuanian text with above accents stack them above U+0307, which is more robust than relying on language-specific glyph processing. It isn’t clear, however, whether Lithuanian text can be relied upon to be encoded in that way, so we tend to provide the locl behaviour as a fallback.

@tiroj The dotted-i with accents is only used in dictionaries and works borrowing dictionary notation, it will not appear in general text. Foreign names are more likely to appear in general text and in fact do appear more often when looking at Lithuanian corpus.

It may make sense for a font designed for Lithuanian dictionaries to use the locl feature to insert a dot on accented i or j but it’s damaging to general text to do so whenever that general text displays place or people names with accented i that is not supposed to have an extra dot. But I guess I’m just unnecessarily repeating myself.

@MariannaPaszkowska @chrissimpkins 
There’s a misunderstanding of what the Lithuanian requirements are. See the list of Lithuanian characters in https://www.unicode.org/L2/L2012/12026r-n4191-lithuanian.pdf.
Lithuanian dictionaries and text borrowing dictionary notation use the following relevant characters:
- i̇̀, i with dot and grave
- i̇́, i with dot and acute
- i̇̃, i with dot and tilde
- į̇́, i with ogonek and dot and acute
- į̇̃ , i with ogonek and dot and tilde
- j̇̃, j with dot and tilde

While these (i̇̀ i̇́ i̇̃ į̇́ į̇̃ j̇̃) may occur without the dots (ì í ĩ į́ į̃ j̃) , having the fallback insertion of the dot breaks people and place names and generalizing this fallback insertion to all combining marks above are problematic.
As mentionned before Dalí with acute will be misrepresented as being Dali̇́ with a dot and acute (when the dictionary pronunciation transcription is actually with a dot and grave).
Furthermore any Latvian names with ī with macron would be displayed with i̇̄ with dot and macron in Lithuanian text with the current implementation when there is no tradition of have a dot and macron in Lithuanian dictionaries.

**Comment by moyogo** [comment]
For what it’s worth, fonts like Arial, Times, Georgia, Verdana, Comic Sans, Courier add a dot to iacute with locl LTH but not on any of the others accent i. This seems to have been a good intention but a poor implementation, probably limited to what is displayed in the Figure 7-2 in Unicode as it only shows iacute whereas Lithuanian references mention the others (i̇̀ i̇́ i̇̃ į̇́ į̇̃ j̇̃).

**Comment by moyogo** [comment]
Just to be clear, with fallback Lithuanian dictionary notation only iacute, igrave, itilde, iogonek acutecomb, iogonek tildecomb, j tildecomb should have an extra dot. Note that itilde is missing, fallback Lithuanian dictionary notation is therefore not currently supported. For fallback Lithuanian dictionary notation, breaking common text with foreign names, the following can be done:
```
sub igrave by i dotaccentcomb gravecomb;
sub iacute by i dotaccentcomb acutecomb;
sub itilde by i dotaccentcomb tildecomb;
sub iogonek' acutecomb by iogonek dotaccentcomb;
sub iogonek' tildecomb by iogonek dotaccentcomb;
sub j' tildecomb by j dotaccentcomb;
```

The glyphs of [soft dotted characters](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=[:Soft_Dotted=Yes:]) in the fonts must loose their dots when followed by a top combining mark. This is regardless whether Lithanian dictionary notation is supported as fallback default feature lookup or as an optional feature lookup. It must always happen.
For example ccmp could have the following, if there is no iogonek.dotless for example:
```
feature ccmp {
sub iogonek' @CombiningTopAccents by idotless ogonekcomb;
sub iogonek' @CombiningNonTopAccents @CombiningTopAccents by idotless ogonekcomb;
sub '[i j] @CombiningTopAccents by @[idotless jdotless];
sub '[i j] @CombiningNonTopAccents @CombiningTopAccents by @[idotless jdotless];
} ccmp;
```

**Comment by MariannaPaszkowska** [comment]
> @MariannaPaszkowska @dberlow re: @moyogo's comment [#33 (comment)](https://github.com/googlefonts/googlesans-flex/issues/33#issuecomment-1342213139) and @tiroj's comment in [#280 (comment)](https://github.com/googlefonts/googlesans-flex/issues/280#issuecomment-1409279465), do we currently support the combining dot above if the text is encoded in this way? I'm trying to understand if this is an issue for our Lithuanian support.

@chrissimpkins yes, we include: u+0307 dotaccentcomb character in GSF. I checked in [this](https://github.com/googlefonts/googlesans-flex/actions/runs/4103075184) build

**Comment by MariannaPaszkowska** [comment]
@moyogo Thanks a lot for the feedback; we'll review the feature code.

**Comment by moyogo** [comment]
>> do we currently support the combining dot above if the text is encoded in this way? I'm trying to understand if this is an issue for our Lithuanian support.

> yes, we include: u+0307 dotaccentcomb character in GSF. I checked in [this](https://github.com/googlefonts/googlesans-flex/actions/runs/4103075184) build

@MariannaPaszkowska  @chrissimpkins Including U+0307 is not enough, [soft dotted](https://www.unicode.org/reports/tr44/#Soft_Dotted) characters must lose their dots when combined with at least one combining top marks.
For example the Unicode recommended /i/dotaccentcomb/acutecomb is currently unchanged with DFLT or with LTH, with two dots instead of one, but it should become /idotless/dotaccentcomb/acutecomb in both, with only one dot.
The dot of the /i that disappears because /i is combined with at least one top mark and the /dotaccentcomb is used here to replace that dot.

Here is what the Unicode recommended `i̇̀ i̇́ i̇̃ į̇́ į̇̃ j̇̃` currently looks like:
![tmp](https://user-images.githubusercontent.com/1923747/217076533-4e29abc7-c9b2-46e2-84c4-bef6b9c46a27.png)

Here is what it should look like (with the ccmp feature code above):
![tmp](https://user-images.githubusercontent.com/1923747/217077152-78cee1a8-e2fa-423d-b6d6-600ed937a3ee.png)

**Comment by moyogo** [comment]
@MariannaPaszkowska @chrissimpkins @tiroj Looking at https://www.unicode.org/L2/L2012/12026r-n4191-lithuanian.pdf, it says:
> In addition, a few scholars have expressed the request that the accent sign on the
letter i should not cancel out the letter’s dot, so that the ordinary characters ì, í, ĩ have been
deemed insufficient, and special variants of accented i’s with both the dot and the accent have
to be introduced.

The "few scholars" seems to indicate not all Lithuanian scholars require this.

Looking at Lithuanian linguistics works this seems to be true. See for example works with [*seniaĩ* or *seniai̇̃*](https://www.google.com/search?tbo=p&tbm=bks&q="seniaĩ"+OR+"seniai̇̃"). There are too many works with ĩ without the extra dot, including metal type works, for them to be mistakes.

Additionally to properly support Lithuanian dictionary notation, all of "Ą́ ą́ Ą̃ ą̃ Ę́ ę́ Ę̃ ę̃ Ė́ ė́ Ė̃ ė̃ i̇̀ i̇́ i̇̃ Į́ į̇́ Į̃ į̇̃ J̃ j̇̃ L̃ l̃ M̃ m̃ R̃ r̃ Ų́ ų́ Ų̃ ų̃ Ū́ ū́ Ū̃ ū̃ " and "Ẽ ẽ Ĩ ĩ Ũ ũ" are required as described in L2-12/026r, n4191.
Handling the softdotted characters i̇̀ i̇́ i̇̃ į̇́ į̇̃ j̇̃ is only a part of it.
It’s not just itilde and a soft dot ccmp feature that are missing but all the other characters and anchor positioning.

Supporting Lithuanian dictionary notation is comparable to supporting English dictionary notations with o͝o, o͞o, etc. or ᵺ, or the English subset of the International Phonetic Alphabet.

**Comment by MariannaPaszkowska** [comment]
Closed with PR #360
